### PR TITLE
Mempool: track total number of allocated bytes

### PR DESCRIPTION
**Pull request by amosr**
! @tranma @jystic

### CODE REVIEW

File: csrc/anemone_mempool.c [hunk 1]
```diff
@@ -36,9 +36,12 @@ void anemone_mempool_add_block (anemone_mempool_t *pool)
 {
   anemone_block_t *next = anemone_block_create (pool->last, anemone_block_size);
 
+  pool->total_alloc_size += anemone_block_size;
   pool->last        = next;
   pool->current_ptr = next->ptr;
   pool->maximum_ptr = next->ptr + anemone_block_size;
+
+  ANEMONE_MEMPOOL_WHEN_DEBUG(fprintf (stderr, "anemone_mempool_add_block: total_alloc_size: %" PRId64 "\n", pool->total_alloc_size));
```
Comment (tranma) [inline csrc/anemone_mempool.c]: oh, is this more accurate than %lld?

Comment (amosr) [inline csrc/anemone_mempool.c]: I think it's that it works on 90s gcc as well as clang

Comment (jystic) [inline csrc/anemone_mempool.c]: yeah it's the technically "correct" way to write "%d" for `int64_t`

File: ctest/test_mempool.c [hunk 2]
```diff
@@ -125,3 +125,37 @@ bool_t test_mempool_nonoverlap(uint64_t iterations, size_t max_bytes)
   return 1;
 }
 
+// Test that the memory pool keeps track of allocated size
+// It won't be exactly the same as iterations*bytes since we are packing into blocks
+bool_t test_mempool_size(uint64_t iterations, size_t bytes)
```
Comment (jystic) [inline ctest/test_mempool.c]: :nextlevel:

**Comment by tranma** [comment]
nice 👍

**Comment by jystic** [comment]
👍 after squashing those wip / fix commits

**Pull request merged**